XamarinCommunityToolkit/Views/CameraView/CameraViewRenderer.uwp.cs

```diff
@@ -0,0 +1,397 @@
+					// TODO
+					break;
+				case nameof(CameraView.Zoom):
+					UpdateZoom();
+					break;
+			}
+			base.OnElementPropertyChanged(sender, e);
+		}
+
+		void UpdateZoom()
+		{
+			var zoomControl = mediaCapture.VideoDeviceController.ZoomControl;
+			if (!zoomControl.Supported)
+				return;
+
+			// Added here since it's an internal method to XF
+			static float Clamp(double value, float min, float max)
+			{
+				if (value.CompareTo(min) < 0)
+					return min;
+				if (value.CompareTo(max) > 0)
+					return max;
+				return (float)value;
+			}
+
+			var settings = new ZoomSettings
+			{
+				//TODO replace clamp
+				Value = Clamp(Element.Zoom, zoomControl.Min, zoomControl.Max),
+				Mode = zoomControl.SupportedModes.Contains(ZoomTransitionMode.Smooth)
+					? ZoomTransitionMode.Smooth
+					: zoomControl.SupportedModes.First()
+			};
+
+			zoomControl.Configure(settings);
+		}
+
+		DeviceInformation FilterCamera(DeviceInformationCollection cameraDevices, Windows.Devices.Enumeration.Panel panel)
+		{
+			foreach (var cam in cameraDevices)
+			{
+				if (cam.EnclosureLocation?.Panel == panel)
+					return cam;
+			}
+			return null;
+		}
+
+		async Task InitializeCameraAsync()
+		{
+			Available = false;
+			if (mediaCapture != null)
+				return;
+
+			var cameraDevices = await DeviceInformation.FindAllAsync(DeviceClass.VideoCapture);
+			if (cameraDevices.Count == 0)
+			{
+				Element?.RaiseMediaCaptureFailed("Camera devices not found.");
+				return;
+			}
+
+			IsBusy = true;
+			DeviceInformation device = null;
+			switch (Element.CameraOptions)
+			{
+				default:
+				case CameraOptions.Default:
+					device = cameraDevices[0];
+					break;
+				case CameraOptions.Front:
+					device = FilterCamera(cameraDevices, Windows.Devices.Enumeration.Panel.Front);
+					break;
+				case CameraOptions.Back:
+					device = FilterCamera(cameraDevices, Windows.Devices.Enumeration.Panel.Back);
+					break;
+				case CameraOptions.External:
+					device = FilterCamera(cameraDevices, Windows.Devices.Enumeration.Panel.Unknown);
+					break;
+			}
+			if (device == null)
+			{
+				Element?.RaiseMediaCaptureFailed($"{Element.CameraOptions} camera not found.");
+				IsBusy = false;
+				return;
+			}
+
+			var audioDevice = await DeviceInformation.FindAllAsync(DeviceClass.AudioCapture);
+			var selectedAudioDevice = audioDevice.Count == 0 ? null : audioDevice[0].Id;
+
+			mediaCapture = new MediaCapture();
+			try
+			{
+				await mediaCapture.InitializeAsync(new MediaCaptureInitializationSettings
+				{
+					VideoDeviceId = device.Id,
+					MediaCategory = MediaCategory.Media,
+					StreamingCaptureMode = selectedAudioDevice == null ? StreamingCaptureMode.Video : StreamingCaptureMode.AudioAndVideo,
+					AudioProcessing = Windows.Media.AudioProcessing.Default,
+					AudioDeviceId = selectedAudioDevice
+				});
+				flash = await Lamp.GetDefaultAsync();
+				if (mediaCapture.VideoDeviceController.ZoomControl.Supported)
```

Comment:
Null reference exception if we click too fast in the `Frontal camera` switch
```suggestion
				if (mediaCapture?.VideoDeviceController.ZoomControl.Supported ?? false)
```

Reply:
Interestingly I get this exception if I toggle Front Camera to fast.

System.Runtime.InteropServices.COMException: 'The request is invalid in the current state.
The request is invalid in the current state.'

Reply:
Sometimes I got this error as well `System.Runtime.InteropServices.COMException`